### PR TITLE
[tests-only][full-ci]Add then step to suites apiSharePublicLink(1 and 2) 

### PR DESCRIPTION
**Pull request by SagarGi**
## Description
This PR adds `Then` steps to existing API test scenarios to better check the results of `When` steps.

**Suites covered:**
- `apiSharePublicLink1`
- `apiSharePublicLink2`

## Related Issue

- Part of https://github.com/owncloud/core/issues/39512

## How Has This Been Tested?

- Locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/Provisioning.php [hunk 1]
```diff
@@ -3483,6 +3483,8 @@ public function userHasBeenAddedToDatabaseBackendGroup(string $user, string $gro
 	 * @param string $group
 	 * @param string|null $method how to add the user to the group api|occ
 	 * @param bool $checkResult if true, then check the status of the operation. default false.
+	 * 							for given step checkResult is expected to be set as true
+	 * 							for when step checkResult is expected to be set as false
```
Comment (SwikritiT) [inline tests/acceptance/features/bootstrap/Provisioning.php:3487]: Why is this indented so far? :thinking:

File: tests/acceptance/features/bootstrap/Provisioning.php [hunk 2]
```diff
@@ -3501,8 +3503,8 @@ public function addUserToGroup(string $user, string $group, ?string $method = nu
 		$method = \trim(\strtolower($method));
 		switch ($method) {
 			case "api":
-				$this->emptyLastHTTPStatusCodesArray();
-				$this->emptyLastOCSStatusCodesArray();
+					$this->emptyLastHTTPStatusCodesArray();
+					$this->emptyLastOCSStatusCodesArray();
```
Comment (SwikritiT) [inline tests/acceptance/features/bootstrap/Provisioning.php:3507]: This too

File: tests/acceptance/features/apiSharePublicLink2/uploadToPublicLinkShare.feature [hunk 3]
```diff
@@ -144,7 +146,7 @@ Feature: upload to a public link share
       | path        | FOLDER |
       | permissions | change |
     And the quota of user "Alice" has been set to "0"
-    Then uploading a file should not work using the <webdav_api_version> public WebDAV API
+    When the public uploads file "test.txt" with content "test-file" using the <webdav_api_version> public WebDAV API
     And the HTTP status code should be "507"
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiSharePublicLink2/uploadToPublicLinkShare.feature:150]:
```suggestion
    Then the HTTP status code should be "507"
```

File: tests/acceptance/features/apiSharePublicLink2/uploadToPublicLinkShare.feature [hunk 4]
```diff
@@ -163,7 +165,7 @@ Feature: upload to a public link share
       | path        | FOLDER |
       | permissions | create |
     And the quota of user "Alice" has been set to "0"
-    Then uploading a file should not work using the <webdav_api_version> public WebDAV API
+    When the public uploads file "test.txt" with content "test-file" using the <webdav_api_version> public WebDAV API
     And the HTTP status code should be "507"
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiSharePublicLink2/uploadToPublicLinkShare.feature:169]:
```suggestion
    Then the HTTP status code should be "507"
```

File: tests/acceptance/features/bootstrap/Provisioning.php (hunk 2 above)
Comment (kiranparajuli589) [inline tests/acceptance/features/bootstrap/Provisioning.php:3507]: fix indent here; maybe revert

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 5]
```diff
@@ -3337,6 +3340,29 @@ public function thePublicAccessesThePreviewOfTheSharedFileUsingTheSharingApi(str
...
+	 *
+	 * @throws Exception
+	 * @return void
+	 */
+	public function thePublicAccessesThePreviewOfTheFollowingSharedFileUsingTheSharingApi(
+		TableNode $table
+	):void {
+		$this->verifyTableNodeColumns($table, ["path"]);
+		$paths = $table->getHash();
+		$this->emptyLastHTTPStatusCodesArray();
```
Comment (kiranparajuli589) [inline tests/acceptance/features/bootstrap/Sharing.php:3359]: for the sharing api, also use the OCS codes

**Review by kiranparajuli589 — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/bootstrap/Provisioning.php (hunk 1 above)
Comment (SagarGi) [reply tests/acceptance/features/bootstrap/Provisioning.php:3487]: @SwikritiT because the comment is for the $checkResult so that it becomes readable.

**Comment by kiranparajuli589** [comment]
build failed at `coding-standard` https://drone.owncloud.com/owncloud/core/34870/3/6

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

**Review by kiranparajuli589 — APPROVED** [review]

File: tests/acceptance/features/apiSharePublicLink1/accessToPublicLinkShare.feature [hunk 6]
```diff
@@ -36,10 +36,11 @@ Feature: accessing a public link share
       | path        | /FOLDER   |
       | permissions | change    |
       | password    | testpass1 |
-    When the public accesses the preview of file "testavatar.jpg" from the last shared public link using the sharing API
-    Then the HTTP status code should be "404"
-    When the public accesses the preview of file "textfile0.txt" from the last shared public link using the sharing API
-    Then the HTTP status code should be "404"
+    When the public accesses the preview of file following files from the last shared public link using the sharing API
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharePublicLink1/accessToPublicLinkShare.feature:39]:
```suggestion
    When the public accesses the preview of the following files from the last shared public link using the sharing API
```

File: tests/acceptance/features/apiSharePublicLink1/accessToPublicLinkShare.feature [hunk 7]
```diff
@@ -49,7 +50,8 @@ Feature: accessing a public link share
     And user "Alice" has created a public link share with settings
       | path        | /FOLDER |
       | permissions | change  |
-    When the public accesses the preview of file "testavatar.jpg" from the last shared public link using the sharing API
-    Then the HTTP status code should be "200"
-    When the public accesses the preview of file "textfile0.txt" from the last shared public link using the sharing API
-    Then the HTTP status code should be "200"
+    When the public accesses the preview of file following files from the last shared public link using the sharing API
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharePublicLink1/accessToPublicLinkShare.feature:53]:
```suggestion
    When the public accesses the preview of the following files from the last shared public link using the sharing API
```

File: tests/acceptance/features/apiSharePublicLink2/updatePublicLinkShare.feature [hunk 8]
```diff
@@ -437,28 +437,23 @@ Feature: update a public link share
...
     And as "Alice" file "PARENT/CHILD/child.txt" should exist
 
     @notToImplementOnOCIS @issue-ocis-2079
     Examples:
-      | ocs_api_version | ocs_status_code | webdav_api_version |
-      | 1               | 100             | old                |
-      | 2               | 200             | old                |
+      | ocs_api_version | webdav_api_version |
+      | 2               | old                |
+      | 1               | old                |
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharePublicLink2/updatePublicLinkShare.feature:450]:
```suggestion
      | 1               | old                |
      | 2               | old                |
```

Future readers will be confused if the order of examples is reversed here - they will wonder if there is some special reason.

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 9]
```diff
@@ -3337,6 +3340,30 @@ public function thePublicAccessesThePreviewOfTheSharedFileUsingTheSharingApi(str
 		$shareData = $this->getLastShareData();
 		$token = (string) $shareData->data->token;
 		$this->getPublicPreviewOfFile($path, $token);
+		$this->pushToLastStatusCodesArrays();
+	}
+
+	/**
+	 * @When the public accesses the preview of file following files from the last shared public link using the sharing API
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/Sharing.php:3347]:
```suggestion
	 * @When the public accesses the preview of the following files from the last shared public link using the sharing API
```

**Review by phil-davis — APPROVED** [review]

**Pull request merged**